graphql/resolvers.go

```diff
@@ -720,6 +720,14 @@ func (r *queryResolver) CommitQueue(ctx context.Context, id string) (*restModel.
 		}
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("error finding commit queue for %s: %s", id, error.Error(err)))
 	}
+	for i, item := range commitQueue.Queue {
+		issue := *item.Issue
+		patch, err := r.sc.FindPatchById(issue)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("error finding patch for %s: %s", issue, error.Error(err)))
+		}
+		commitQueue.Queue[i].Patch = patch
+	}
```

Comment:
this will make a request to DB for every item in commit queue, which could potentially be dozens (hundreds, thousands?). So that this query does not take a long time, we will want to do as little db requests in the resolvers as possible. You can accomplish this by getting all patches that match the ids in commit queue. 

there is a patch method in `patch/db.go` that will find patches based on query. I do not see a `ByIds` method in that file so you may have to make one. but @john-m-liu would know this better than I. 

this also brings up the point of pagination. this resolver quickly becomes less and less performant for every additional patch in the queue.

Reply:
Yep I would add a db query that takes in multiple IDs and does the query once, something like https://github.com/evergreen-ci/evergreen/blob/master/model/host/db.go#L556

I don't think pagination is necessary, we won't realistically get a commit queue that is more than 20 items deep

Reply:
is it possible to find out the max number of items that has ever been in the commit queue at once? pagination makes the front end resilient to the rare circumstances where expectations (20 items) is far exceeded by reality (80 or 100 items). 

it seems fair to make it a standard that all dynamic lists of items are paginated. this way, we can request 20 or 30 items at a time if we want. and then if there are more than that, we can still be sure our app is fast and adapts to the increase in data.

Reply:
The historical data isn't maintained that I'm aware of, but anecdotally it was around 20. Practically speaking since each item is one commit in a repo, it's unlikely to surpass that day when the mongodb server had heavy commit volume and we had slow processing in the commit queue

Reply:
i still think it's a good practice to implement pagination for dynamic lists, especially since it's not hard for our app. but i take your word for it and agree that this feature should not be blocked by pagination. 

what i would like to do tho is measure the latency of the query as it is now (not paginated) against the same query but paginated. if the difference is >= 100ms (or even 50ms tbh) then i would consider it material and push for pagination.